src/Collection/Iterator/BufferedIterator.php

```diff
@@ -214,4 +228,20 @@ public function unserialize($buffer): void
         $this->_started = true;
         $this->_finished = true;
     }
+
+    /**
+     * Magic method used to rebuild the iterator instance.
+     *
+     * @param array $data Data array.
+     * @return void
+     */
+    public function __unserialize(array $data): void
+    {
+        $this->__construct([]);
+        foreach ($data as $value) {
+            $this->_buffer->push($value);
```

Comment:
I am not 100% sure about this. There are no tests for serialization/unserialization in `BuffereredIteratorTest.php`.

Reply:
Does anything serialize it as a side-effect?

Reply:
Codecov said there was no coverage for these lines, so I doubt it. Anyway, I added a test.